modules/planning/tasks/deciders/open_space_fallback_decider.cc

```diff
@@ -22,13 +22,125 @@
 
 namespace apollo {
 namespace planning {
-
 using apollo::common::Status;
+using apollo::common::math::Box2d;
+using apollo::common::math::Vec2d;
+using apollo::common::TrajectoryPoint;
+using apollo::planning::TrajGearPair;
 
 OpenSpaceFallbackDecider::OpenSpaceFallbackDecider(const TaskConfig& config)
     : Decider(config) {}
 
-Status OpenSpaceFallbackDecider::Process(Frame* frame) { return Status::OK(); }
+Status OpenSpaceFallbackDecider::Process(Frame* frame) {
+    BuildPredictedEnvironment(frame->obstacles());
+
+    if (!IsCollisionFreeTrajectory(
+      frame->open_space_info().chosen_paritioned_trajectory())) {
+      // change gflag
+      frame_->mutable_open_space_info()->set_fallback_flag(true);
+
+      // generate fallback trajectory base on current partition trajectory
+      // vehicle speed is decreased to zero inside safety distance
+      frame_->mutable_open_space_info()->set_fallback_trajectory(
```

Comment:
use:
const auto &open_space_info = frame_->_open_space_info()
auto* open_space_info_ptr = frame_->mutable_open_space_info()
to make codes shorter